src/lib.rs

```diff
@@ -31,17 +34,33 @@ pub enum Wildcard {
     Null,
 }
 
+impl Wildcard {
+    /// Check whether this wildcard matches some JSON value
+    pub fn matches(&self, data: &Value) -> bool {
+        match *self {
+            Wildcard::String => data.is_string(),
+            Wildcard::Number => data.is_number(),
+            Wildcard::Object => data.is_object(),
+            Wildcard::Boolean => data.is_boolean(),
+            Wildcard::Array => data.is_array(),
+            Wildcard::Null => data == &Value::Null,
+        }
+    }
+}
+
 /// Determine if a string represents a wildcard.
 ///
 /// If so, returns Some. Otherwise, returns None.
-pub fn type_wildcard(s: &str) -> Option<Wildcard> {
+pub fn type_wildcards(s: &str) -> Option<Vec<Wildcard>> {
+    // FIXME: Result with error messages
+    // FIXME: parse multiple wildcards
     match s {
-        "[[string]]" => Some(Wildcard::String),
-        "[[number]]" => Some(Wildcard::Number),
-        "[[object]]" => Some(Wildcard::Object),
-        "[[boolean]]" => Some(Wildcard::Boolean),
-        "[[array]]" => Some(Wildcard::Array),
-        "[[null]]" => Some(Wildcard::Null),
+        "[[string]]" => Some(vec![Wildcard::String]),
+        "[[number]]" => Some(vec![Wildcard::Number]),
+        "[[object]]" => Some(vec![Wildcard::Object]),
+        "[[boolean]]" => Some(vec![Wildcard::Boolean]),
+        "[[array]]" => Some(vec![Wildcard::Array]),
+        "[[null]]" => Some(vec![Wildcard::Null]),
```

Comment:
Why does this return a vector now? Is it because we will later implement multi-type wildcards?